src/config.c

```diff
@@ -24,7 +25,7 @@ static bool _initialized = false;
 // command line arguments if specified
 // otherwise returns thread count equals to the number
 // of cores available
-static long long _Config_GetThreadCount(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+static long long _Config_SetThreadCount(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
```

Comment:
Set functions may return the set value but they should actually perform the set.
the process of looking up a config parameter, e.g. `THREAD_COUNT` should be placed in a utility function within this file.